hbase-common/src/main/java/org/apache/hadoop/hbase/MetaTableName.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@InterfaceAudience.Public
+public class MetaTableName {
+  private static final Logger LOG = LoggerFactory.getLogger(MetaTableName.class);
+  private static volatile TableName instance;
+
+  private MetaTableName() {
+  }
+
+  /**
+   * Get the singleton instance of the meta table name.
+   * @return The meta table name instance
+   */
+  public static TableName getInstance() {
+    if (instance == null) {
+      synchronized (MetaTableName.class) {
+        if (instance == null) {
+          instance = initializeHbaseMetaTableName(HBaseConfiguration.create());
```

Comment:
I do not think this is a good practise, as this is still static, what if a client wants to connect different read replica clusters? And it will difficult to write UTs when you want to set up two clusters with different meta table names...